src/nlua_shiplog.c

```diff
@@ -0,0 +1,149 @@
+#include "naev.h"
+
+#include <stdlib.h>
+#include <stdio.h>
+#include "nstring.h"
+#include <math.h>
+
+#include <lua.h>
+#include <lauxlib.h>
+
+#include "nlua.h"
+#include "nlua_hook.h"
+#include "nlua_player.h"
+#include "nlua_tk.h"
+#include "nlua_faction.h"
+#include "nlua_system.h"
+#include "nlua_tex.h"
+#include "nlua_camera.h"
+#include "nlua_music.h"
+#include "nlua_bkg.h"
+#include "nlua_misn.h"
+#include "player.h"
+#include "mission.h"
+#include "log.h"
+#include "rng.h"
+#include "toolkit.h"
+#include "land.h"
+#include "nxml.h"
+#include "nluadef.h"
+#include "music.h"
+#include "gui_osd.h"
+#include "npc.h"
+#include "array.h"
+#include "ndata.h"
+#include "shiplog.h"
+
+int shiplog_loadShiplog( nlua_env env );
+static int shiplog_createLog( lua_State *L );
+static int shiplog_appendLog( lua_State *L );
+static const luaL_Reg shiplog_methods[] = {
+   { "createLog", shiplog_createLog },
+   { "appendLog", shiplog_appendLog },
+   {0,0}
+}; /**< Shiplog Lua methods. */
+
+
+/**
+ * @brief Registers the lua libraries.
+ *
+ *    @param L Lua state.
+ *    @return 0 on success.
+ */
+int shiplog_loadLibs( nlua_env env )
+{
+   nlua_loadStandard(env);
+   nlua_loadShiplog(env);
+   return 0;
+}
+/*
+ * individual library loading
+ */
+/**
+ * @brief Loads the mission Lua library.
+ *    @param L Lua state.
+ */
+int nlua_loadShiplog( nlua_env env )
+{
+   nlua_register(env, "shiplog", shiplog_methods, 0);
+   return 0;
+}
+
+
+/**
+ * @brief Creates a shiplog for this mission.
+ *
+ *    @luatparam string idstr ID string to identify this log, or empty string for unnamed logsets.
+ *    @luatparam string name Name for this log.
+ *    @luatparam string type Type of log (e.g travel, trade, etc, can be anything.)
+ *    @luatparam boolean overwrite Whether to overwrite existing mission with this logname and logtype.  Warning, removes previous entries of this logname and type.
+ *    @luatparam int maxLen Maximum length of the log (zero for infinite) - if greater than this length, new entries appended will result in old entries being removed.
+ *
+ *    @luatreturn number The logid of the mission.
+ * @luafunc createLog( idstr, logname, logtype, overwrite, maxLen )
+ * @usage misn.createLog("MyID","My mission title","Mission type",false, 0) -- where myID can be a string for this mission set (e.g. "shadow"), or an empty string.
+ * @usage misn.createLog("","Any title","Anything can be a type",true, 10) -- with true to replace existing missions of this title and type, and 10 to limit this log to 10 entries (pruning when new ones are added beyond this).
+ */
+static int shiplog_createLog( lua_State *L )
+{
+   const char *logname;
+   const char *logtype;
+   const char *nidstr;
+   char *idstr=NULL;
+   int overwrite,logid,maxLen;
+   /* Parameters. */
+   nidstr      = luaL_checkstring(L,1);
+   logname    = luaL_checkstring(L,2);
+   logtype    = luaL_checkstring(L,3);
+   overwrite = lua_toboolean(L,4);
+   maxLen = luaL_checkint(L,5);
```

Comment:
I believe (to do what I commented on the doc line), you would do something like:

```
maxLen = 0;
if ( lua_gettop(L) > 4 )
   maxLen = MAX(0, luaL_checkint(L,5));
```

Reply:
Thanks - I've added this.